### PR TITLE
Added save_as_msp function.

### LINKED ISSUE #166: Would be nice to have a save_as_msp module
Hi All, Hi @florian-huber 

matchms is a really nice tools for the community, Thanks.
I would like to know if you plan to create a save_as_msp module?

I'm asking because I'm using NIST database a lot and it would be nice to be able to create MSP database from GNPS/MSHUB MGF outputs.

Best
Yann

### PR DESCRIPTION
**Pull request by hechth**
Implemented save_as_msp function and test cases for both msp files contained in test data.

Closes #166

### CODE REVIEW

**Comment by hechth** [comment]
Running `pytest` locally reports 100% coverage on the save_as_msp.py file - I'd be happy to extend the testing though in case the real coverage is really only 33%.

File: tests/test_save_as_msp.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        filename = os.path.join(temp_dir, "test.mzml")
+
+        with pytest.raises(AssertionError) as exception:
+            save_as_msp(None, filename)
+
+        message = exception.value.args[0]
+        assert message == "File extension must be 'msp'."
+
+
+# Using tmp_path fixture from pytest: https://docs.pytest.org/en/stable/tmpdir.html#the-tmp-path-fixture
```
Comment (florian-huber) [inline tests/test_save_as_msp.py:58]: Nice use of fixture in this test file!

**Review by florian-huber — APPROVED** [review]
Nice addition @hechth! And nice, clean code + tests, so thanks a lot!

(I didn't have much time recently, so it took me a while to fix the conda issues, should now be done with --> #217)

**Comment by hechth** [comment]
@florian-huber I've seen the changes you made in #217 which should fix the failing check after merging it into the main branch - should I copy/pull these fixes into this PR to pass all checks or is it okay to merge without that step?

**Comment by florian-huber** [comment]
> should I copy/pull these fixes into this PR to pass all checks or is it okay to merge without that step?

I think in this case both would work, but I prefer first merging #217 into this one (saw that you reviewed it so I will simply merge it and once the tests here pass we can go ahead and merge this one, too).

**Comment by florian-huber** [comment]
@hechth , I just merged #217 into this PR and added `save_as_msp` to `CHANGELOG.md`. To me it's now ready to be merged (once tests pass).

**Pull request merged**